data_steward/cdr_cleaner/manual_cleaning_rules/dc732_suppress.py

```diff
@@ -0,0 +1,445 @@
+    return pd.concat(CLIENT.query(dataset_query).to_dataframe() for dataset_query in dataset_queries)
+
+
+def ppi_pm(g):
+    """
+    True if any rows in the group have source PPI/PM
+    """
+    return g['source'].isin(['PPI/PM']).any()
+
+
+def get_suppress_summary_df(rows_to_suppress_df):
+    """
+    Retrieve summary info (counts) on rows to suppress from target datasets
+
+    :param rows_to_suppress_df: info on rows to suppress from target datasets
+    :return: dataframe with dataset_id, table, disallowed_concept_id, person_id, row_id, source
+    """
+    summary_cols = ['dataset_id', 'table', 'disallowed_concept_id', 'person_id', 'row_id', 'source']
+    group_by_concept = rows_to_suppress_df[summary_cols].groupby(['dataset_id', 'table', 'disallowed_concept_id'])
+    summary_df = group_by_concept.nunique().drop(columns=['dataset_id', 'table', 'disallowed_concept_id'])
+    summary_df = summary_df.add_prefix('n_')
+    summary_df['ppi_pm'] = group_by_concept.apply(ppi_pm)
+    return summary_df.sort_values(by=['dataset_id', 'table', 'n_person_id'], ascending=False)
+
+
+# # Backup Rows
+# Backup rows are stored in tables whose structure is identical to domain tables except for the addition of fields to distinguish the dataset and table the original row came from. For example, `visit_occurrence` rows would be backed up in a table with the below schema.
+#
+# | Field name                    | Type    |
+# |-------------------------------|---------|
+# | **dataset_id**                | STRING  |
+# | **table**                     | STRING  |
+# | visit_occurrence_id           | INTEGER |
+# | person_id                     | INTEGER |
+# | visit_concept_id              | INTEGER |
+# | --- _other columns_  ---      |         |
+# | discharge_to_source_value     | STRING  |
+# | preceding_visit_occurrence_id | INTEGER |
+
+# +
+BACKUP_TABLE_ROWS_QUERY = JINJA_ENV.from_string("""
+{% for dataset_id in dataset_ids %}
+SELECT 
+ '{{dataset_id}}' AS dataset_id, 
+ '{{table}}' table, 
+ t.* 
+FROM {{dataset_id}}.{{table}} t
+JOIN {{suppress_rows_table}} r
+ ON r.row_id = t.{{table}}_id
+WHERE r.dataset_id = '{{dataset_id}}'
+AND r.table = '{{table}}'
+{% if loop.nextitem is defined %}
+ UNION ALL
+{% endif %}
+{% endfor %}
+""")
+
+BACKUP_EXT_TABLE_ROWS_QUERY = JINJA_ENV.from_string("""
+{% for dataset_id in dataset_ids %}
+SELECT 
+ '{{dataset_id}}' AS dataset_id, 
+ '{{table}}_ext' table, 
+ t.* 
+FROM {{dataset_id}}.{{table}}_ext t
+JOIN {{suppress_rows_table}} r
+ ON r.row_id = t.{{table}}_id
+WHERE r.dataset_id = '{{dataset_id}}'
+AND r.table = '{{table}}'
+{% if loop.nextitem is defined %}
+ UNION ALL
+{% endif %}
+{% endfor %}
+""")
+
+
+def backup_rows_to_suppress(tables_to_suppress_df, suppress_rows_table):
+    """
+    Create a table for each domain and load rows that will be suppressed across all datasets
+    Note: Rows are appended if tables exist
```

Comment:
nit:  add blank line between one line summary and this line.  if/when we use documentation tools, this will make a difference.

Reply:
Thanks, I wasn't sure and could not find a pydoc reference with guidance on how to put notes.

Reply:
i think this is just a multi-line docstring convention.  https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings  

i used pep guidance as the source for the suggestion.